File: src/main/java/main/utils/lavalink/LavaNode.java
package main.utils.lavalink;

import lombok.Getter;

import java.net.URI;

public class LavaNode {
    private final String host;
    private final String port;
    @Getter
    private final String password;

    public LavaNode(String host, String port, String password) {
        this.host = host;
        this.port = port;
        this.password = password;
    }

    public URI getURI() {
        return URI.create("ws://" + host + ((!port.isEmpty() && !port.isBlank()) ? ":" + port : ""));
    }

    @Override
    public String toString() {
        return "LavaNode{" +
                "host='" + host + '\'' +
                ", port='" + port + '\'' +
                ", password='" + password + '\'' +
                '}';
    }
}

File: src/main/kotlin/main/utils/lavalink/LavaNode.kt
package main.utils.lavalink

import java.net.URI

data class LavaNode(
    val name: String = "Lava Node",
    private val host: String,
    private val port: String,
    val password: String
) {

    val uri: URI
        get() = URI.create("ws://$host${ if (port.isNotEmpty() && port.isNotBlank()) ":$port" else ""}")


    override fun toString(): String =
        "LavaNode{name ='$name' ,host='$host', port='$port', password='$password'}"

}
